crates/shared/src/lib.rs

```diff
@@ -141,6 +141,17 @@ macro_rules! shared_api {
             generate_typescript: bool,
             generate_jsdoc: bool,
             variadic: bool,
+            fn_attrs: Option<FunctionAttributes>,
+        }
+
+        struct FunctionAttributes {
+            ret: FunctionComponentAttributes,
+            args: Vec<FunctionComponentAttributes>,
```

Comment:
Lets fold this into `arg_names` again.
The custom return type and description can be split into two fields instead of the struct which won't be reused with args anymore.

Reply:
No two individual fields. No reason to make a new struct if its only used in a single place.